### PR TITLE
add sorting function

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Adds sorting buttons to the columns of the CCP endpoint treatment options table.  Clicking on the patient icon, or the 1 yr, 3 yr, or 5 yr headers will sort the rows based on the info in that column.  Clicking once sorts it in ascending order, clicking again sorts in descending, and clicking a third time resets to normal.  Normal is alphabetically sorted based on the display name.  

The designs show a downward facing arrow, which is descending _from the top_.  The way I implemented it a downward facing arrow is descending _from the bottom_.  To me having the upward arrow represent ascending values (from the bottom) is more natural.  This is a simple fix, though, and can be easily changed to what [the designs](https://www.figma.com/proto/rLpubNAfrqV99exbLVFz5c/core-cancer-pilot?node-id=2507%3A13716&viewport=-533%2C-2626%2C0.43955686688423157&scaling=min-zoom&redirected=1) show.

Some of the tests had to be edited because they check the treatment options in a certain order and the alphabetical sorting caused errors.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [x] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [x] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/mcode-pilot/components/TreatmentOptionsOutcomesTable/TreatmentOptionsOutcomesTable.js [hunk 1]
```diff
@@ -198,6 +240,32 @@ export default class TreatmentOptionsOutcomesTable extends Component {
         );
     }
 
+    sortTreatments(propName) {
```
Comment (dehall) [inline src/mcode-pilot/components/TreatmentOptionsOutcomesTable/TreatmentOptionsOutcomesTable.js]: The function name here is a little misleading because this isn't actually sorting the treatments, this gives you the comparator to use for sorting treatments. Can you rename this so it's more clear on what it's doing?

**Comment by jafeltra** [comment]
This looks good to me! I personally like the ordering of the sort as it is now, and since it now lines up with the mockups, I think it's good. One small thing:

[JA1] The cursor doesn't change to indicate that you can click on the table header/arrows. Can you make it so it's the pointer to indicate that it can be clicked on?

**Comment by noranda** [comment]
This looks good!

**Comment by noranda** [comment]
@KeeyanGhoreshi This just needs to be rebased and it's good to go.

**Comment by jafeltra** [comment]
JA1 looks good to me!
